docs/transparent-validation/supermicro_x11_lga1151_series/theory-of-operation.md: add file (#585)
Dasharo/docs: docs/transparent-validation/supermicro_x11_lga1151_series/theory-of-operation.md

# Supermicro X11 LGA1151 Series theory of operation

## Remote power control using ipmitool

To remote control supermicro you can use ipmitool.

```bash

> ipmitool -H [bmc_ip] -U [bmc_username] -P [bmc_password] chassis power off

Reviewer: Where should I look for credentials?

Reviewer 2: Could be a good idea to store them in snipeit, along with the other information, such as IP addresses?

Author: @krystian-hebel I added credentiale on snipeit

ipmitool -H [bmc_ip] -U [bmc_username] -P [bmc_password] chassis power on
ipmitool -H [bmc_ip] -U [bmc_username] -P [bmc_password] chassis power cycle
ipmitool -H [bmc_ip] -U [bmc_username] -P [bmc_password] chassis power reset
```

## Enabling SOL

This procedure is based on
[this documentation](https://www.fmad.io/blog/supermicro-serial-kvm).

Follow the steps below to set up a serial connection via ssh:

1. Enable SOL/COM2 console in BIOS. To do that in BIOS go to the `Advanced`
   section, then select `Serial Port Console Redirection` and set the
   `SOL/COM2 Console Redirection` option to `Enabled` state.

1. Edit the `/etc/default/grub` on your Linux-based OS file by adding:

    ```bash
    linux /vmlinuz ro console=ttyS1,115200n8 earlyprintk=serial,ttyS1,115200n8
    ```

1. Connect with BMC via ssh:

    ```bash

>     ssh ADMIN@<IP>

Reviewer 2: It should work via password authentication? It requires a key it appears. And documentation does not mention how to add one.

Author: I prepared the documentation on a system with OpenSSH < 8.8, and it worked without any configuration.
https://github.com/Dasharo/docs/pull/585/commits/53c1c25d09798d99eef8d08b88cf1fea0e20f2c4 - I added the missing step

Author: @macpijan ping

Reviewer 3: @krystian-hebel any ideas?

Reviewer 2: The updated documentation worked for me.

# Supermicro X11 LGA1151 Series theory of operation

## Remote power control using ipmitool

To remote control supermicro you can use ipmitool.

```bash
ipmitool -H [bmc_ip] -U [bmc_username] -P [bmc_password] chassis power off
ipmitool -H [bmc_ip] -U [bmc_username] -P [bmc_password] chassis power on
ipmitool -H [bmc_ip] -U [bmc_username] -P [bmc_password] chassis power cycle
ipmitool -H [bmc_ip] -U [bmc_username] -P [bmc_password] chassis power reset
```

## Enabling SOL

This procedure is based on
[this documentation](https://www.fmad.io/blog/supermicro-serial-kvm).

Follow the steps below to set up a serial connection via ssh:

1. Enable SOL/COM2 console in BIOS. To do that in BIOS go to the `Advanced`
   section, then select `Serial Port Console Redirection` and set the
   `SOL/COM2 Console Redirection` option to `Enabled` state.

1. Edit the `/etc/default/grub` on your Linux-based OS file by adding:

>     ```bash

Reviewer: ```suggestion
    ```text
```

    linux /vmlinuz ro console=ttyS1,115200n8 earlyprintk=serial,ttyS1,115200n8
    ```

    > NOTE: If your Linux-based OS is QubesOS you should additionally remove the
      `plymouth.ignore-serial-consoles` phase to enable password disk
      authentication via SOL

> 1. Add the following lines to the `/etc/ssh/ssh_config` file on your machine:
> 
>     ```bash
>     HostKeyAlgorithms = +ssh-rsa
>     PubkeyAcceptedAlgorithms = +ssh-rsa
>     ```

Reviewer: Adding such settings to global configuration is asking for problems...

```suggestion
1. Add the following lines to the `~/.ssh/config` file on your machine:

    ```text
    Host <IP>
    	HostKeyAlgorithms = +ssh-rsa
    	PubkeyAcceptedAlgorithms = +ssh-rsa
    ```
```

1. Connect with BMC via ssh:

    ```bash
    ssh ADMIN@<IP>
    ```

1. Then run the following command:

    ```bash
    cd system1/sol1
    ```

1. And start the serial console:

    ```bash
    start
    ```

> If you want to run the SOL console again, repeat points 3-5.

Reviewer: ```suggestion
If you want to run the SOL console again, repeat points 4-6.
```
